src/cls/rgw/cls_rgw.cc

```diff
@@ -3168,18 +3179,10 @@ static int gc_omap_get(cls_method_context_t hctx, int type, const string& key, c
   string index;
   prepend_index_prefix(key, type, &index);
 
-  bufferlist bl;
-  int ret = cls_cxx_map_get_val(hctx, index, &bl);
+  int ret = read_omap_entry(hctx, key, info);
```

Comment:
`key` -> `index`?

Reply:
guess this is indeed the source of failures, thanks!